src/Core/Framework/App/AppUrlChangeResolver/UninstallAppsStrategy.php

```diff
@@ -72,7 +68,7 @@ public function resolve(Context $context): void
         $apps = $this->appRepository->search(new Criteria(), $context)->getEntities();
 
         foreach ($apps as $app) {
-            $this->themeLifecycleHandler->handleUninstall(new AppDeactivatedEvent($app, $context));
+            $this->appStateService->deactivateApp($app->getId(), $context);
```

Comment:
There was a reason we deleted the apps manually here, because we should not notify the app that it get's deactivated in this shop, because the app is still running in another shop on a different URL, when dispatching the app deactivated Event, we would also dispatch the webhooks to the app which will lead into errors

Reply:
the easiest solution would be to add `on-invalid="null"` to the service definition for the lifecycle handler and perform a null check before calling that class